src/error.rs

```diff
@@ -0,0 +1,14 @@
+use std::str;
+
+#[derive(Debug)]
```

Comment:
I'm a strong proponent of us using `failure` throughout this crate for error handling/management

Reply:
I think `failure` is great for code that's doing a lot of handling of various dependencies from various sources. It's super annoying to write all that code manually. I'm using `failure` myself in applications. But for this crate I don't think we'd be needing it very much... beyond utf-8 conversion errors and I/O stuff I don't think we'd be having many different error sources.

And failure does come with a cost: it has quite many dependencies. This cost is paid in large applications, but I want to keep this crate as universally usable as possible :). So I'd prefer to go without failure.

There is one advantage of failure, which is backtraces, but we can offer those behind a feature flag or something. Then users who prefer a minimal experience can just not enable that flag. IMO backtraces are better done in a separate PR.